tiny_dnn/layers/deconvolutional_layer.h

```diff
@@ -198,7 +214,12 @@ class deconvolutional_layer : public layer {
                         size_t w_stride        = 1,
                         size_t h_stride        = 1,
                         backend_t backend_type = core::default_engine())
-    : layer(std_input_order(has_bias), {vector_type::data}) {
+    : layer({vector_type::data}, {vector_type::data}) {
+    layer::add_parameter(out_channels, in_channels, window_height, window_width,
+                         parameter_type::weight);
```

Comment:
actually, I think that whatever data allocation done in `layer({vector_type::data}, {vector_type::data})` should happen when you add a parameter to the layer. Additionally, we should be able to control whether we use imported wieghts. It's not clear to me what this `add_parameter(...)` does: allocates data?